tools/statplot.py

```diff
@@ -204,14 +225,28 @@ def KeyPressed(self, widget, event):
             self.PlotType = 'line' if self.PlotType == 'marker' else 'marker'
             self.PlotData('update', self.PlotType, self.ax.get_xscale(),
                           self.ax.get_yscale())
+        elif key == 'a':
+            if (self.xCombo.get_child().has_focus()
+                    or self.yCombo.get_child().has_focus()):
+                pass
+            else:
+                self.xCombo.grab_focus()
+        elif key == 'o':
+            if (self.xCombo.get_child().has_focus()
+                    or self.yCombo.get_child().has_focus()):
+                pass
+            else:
+                self.yCombo.grab_focus()
 
     def PlotData(self, action, type, xscale, yscale):
-        self.xField = self.xCombo.get_active_text()
-        self.yField = self.yCombo.get_active_text()
-        self.xData = self.entries[self.xField]
-        self.yData = self.entries[self.yField]
-        if numpy.unique(self.xData).size == 1 \
-                and numpy.unique(self.yData).size == 1:
+        if len(self.values.shape) == 1:
+            self.xData = self.values[self.xCombo.get_active()]
+            self.yData = self.values[self.yCombo.get_active()]
+        else:
+            self.xData = self.values[:, self.xCombo.get_active()]
+            self.yData = self.values[:, self.yCombo.get_active()]
```

Comment:
I think you can tidy this up with:

```python
self.xData = self.values[..., self.xCombo.get_active()]
```

Should work for any number of dimensions

Reply:
Thanks for showing an example use of Ellipsis objects, never really tried to understand what they are before!